velox/functions/prestosql/aggregates/MaxSizeForStatsAggregate.cpp

```diff
@@ -0,0 +1,239 @@
+          result = value;
+        },
+        mayPushdown,
+        (int64_t)0);
+  }
+
+  void addSingleGroupRawInput(
+      char* group,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdateSingleGroup(group, rows, args[0]);
+  }
+
+  void addRawInput(
+      char** groups,
+      const SelectivityVector& rows,
+      const std::vector<VectorPtr>& args,
+      bool /*mayPushdown*/) override {
+    doUpdate(groups, rows, args[0]);
+  }
+
+ protected:
+  void upsertOneAccumulatorNew(
+      char* const group,
+      std::vector<vector_size_t>& rowSizes,
+      vector_size_t idx) {
+    if (decoded_.isNullAt(idx)) {
+      return;
+    }
+
+    // Clear null.
+    clearNull(group);
+    // Set max(current, this).
+    int64_t& current = *value<int64_t>(group);
+    current = std::max(current, (int64_t)rowSizes[idx]);
+  }
+
+  void
+  doUpdate(char** groups, const SelectivityVector& rows, const VectorPtr& arg) {
+    decoded_.decode(*arg, rows, true);
+
+    if (decoded_.isConstantMapping() && decoded_.isNullAt(0)) {
+      // There's nothing to do; all values are null.
+      return;
+    }
+
+    if (decoded_.isConstantMapping()) {
+      estimateSerializedSizes(arg, 1);
+      std::fill(elementSizes_.begin(), elementSizes_.end(), elementSizes_[0]);
+    } else {
+      estimateSerializedSizes(arg, arg->size());
+    }
+    rows.applyToSelected([&](vector_size_t i) {
+      upsertOneAccumulatorNew(groups[i], elementSizes_, i);
+    });
+  }
+
+  // Estimate the sizes of first numRows elements in vector.
+  void estimateSerializedSizes(VectorPtr vector, vector_size_t numRows) {
+    elementSizes_.resize(numRows);
+    std::fill(elementSizes_.begin(), elementSizes_.end(), 0);
+    elementSizePtrs_.resize(numRows);
+    elementIndices_.resize(numRows);
+
+    for (int i = 0; i < numRows; i++) {
+      elementIndices_[i] = IndexRange{i, 1};
+      elementSizePtrs_[i] = &elementSizes_[i];
+    }
+
+    getVectorSerde()->estimateSerializedSize(
+        vector,
+        folly::Range(elementIndices_.data(), numRows),
+        elementSizePtrs_.data());
+  }
+  void doUpdateSingleGroup(
+      char* group,
+      const SelectivityVector& rows,
+      const VectorPtr& arg) {
+    decoded_.decode(*arg, rows, true);
+
+    if (decoded_.isConstantMapping()) {
+      if (decoded_.isNullAt(0)) {
+        // There's nothing to do; all values are null.
+        return;
+      }
+      // Estimate first element because it is constant mapping.
+      estimateSerializedSizes(arg, 1);
+      upsertOneAccumulatorNew(group, elementSizes_, 0);
+    }
+
+    estimateSerializedSizes(arg, arg->size());
+    rows.applyToSelected([&](vector_size_t i) {
+      upsertOneAccumulatorNew(group, elementSizes_, i);
+    });
+  }
+};
+
+bool registerMaxSizeForStatsAggregate(const std::string& name) {
+  // Types here are used by PlanBuilder to populate partial and final
```

Comment:
As we discussed earlier, this comments probably doesn't belong to an aggregate function implementation, but rather somewhere in Aggregate.h or in documentation. Also, PlanBuilder is a test-only utility which represents just one usage of the signatures.